test/e2e-wi-tests.sh

```diff
@@ -0,0 +1,186 @@
+    service_account_key="${PUBSUB_SERVICE_ACCOUNT_KEY}"
+  fi
+}
+
+# Tear down resources required for Control Plane setup.
+function control_plane_teardown() {
+  # When not running on Prow we need to delete the service accounts and namespaces created.
+  if (( ! IS_PROW )); then
+    echo "Tear down ServiceAccount for Control Plane"
+    gcloud iam service-accounts keys delete -q ${CONTROL_PLANE_SERVICE_ACCOUNT_KEY} \
+      --iam-account=${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/pubsub.admin
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/pubsub.editor
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/storage.admin
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/cloudscheduler.admin
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/logging.configWriter
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/logging.privateLogViewer
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/cloudscheduler.admin
+    # Remove iam.serviceAccountAdmin role to the Google service account.
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/iam.serviceAccountAdmin
+    gcloud iam service-accounts remove-iam-policy-binding \
+      --role roles/iam.workloadIdentityUser \
+      --member ${MEMBER} ${AUTHENTICATED_SERVICE_ACCOUNT}
+  else
+    gcloud iam service-accounts remove-iam-policy-binding \
+      --role roles/iam.workloadIdentityUser \
+      --member ${MEMBER} \
+      --project ${PROW_PROJECT_NAME} ${AUTHENTICATED_SERVICE_ACCOUNT}
+  fi
+}
+
+# Create a cluster with Workload Identity enabled.
+initialize $@ --cluster-creation-flag "--workload-pool=\${PROJECT}.svc.id.goog"
+
+kubectl annotate serviceaccount ${K8S_CONTROLLER_SERVICE_ACCOUNT} iam.gke.io/gcp-service-account=${AUTHENTICATED_SERVICE_ACCOUNT} \
```

Comment:
Move this command call to `control_plane_setup`?

Reply:
These two comand? annotate and initialize?

Reply:
No just this annotate command (please note `control_plane_setup` is also called after `initialize`)

Reply:
I see, so I need to add initialize at the beginning of control_plane_setup? and annotate at the end?

Reply:
Sorry, ignore my last commend...